diff --git a/images/linux/scripts/installers/1804/dotnetcore-sdk.sh b/images/linux/scripts/installers/1804/dotnetcore-sdk.sh
index f779ff5..f0ef8d5 100644
--- a/images/linux/scripts/installers/1804/dotnetcore-sdk.sh
+++ b/images/linux/scripts/installers/1804/dotnetcore-sdk.sh
@@ -56,6 +56,20 @@ done

 sortedSdks=$(echo ${sdks[@]} | tr ' ' '\n' | grep -v preview | grep -v rc | grep -v display | cut -d\" -f2 | sort -u -r)

+partialSdks=$(echo "${sortedSdks}" | cut -d"." -f1-2 | uniq)
+finalSdks=""
+
+for partialSdk in $partialSdks; do
+  for fullSdk in $sortedSdks; do
+    if [[ $fullSdk == $partialSdk* ]]; then
+      finalSdks+="$fullSdk "
+      break
+    fi
+  done
+done
+
+sortedSdks="${finalSdks}"
+
 for sdk in $sortedSdks; do
     url="https://dotnetcli.blob.core.windows.net/dotnet/Sdk/$sdk/dotnet-sdk-$sdk-linux-x64.tar.gz"
     echo "$url" >> urls
